src/name.rs

```diff
@@ -806,6 +813,44 @@ mod namespaces {
         assert_eq!(resolver.find(name, &buffer), Unknown(b"unknown".to_vec()));
     }
 
+    #[test]
+    fn undeclared_xml_prefix() {
```

Comment:
I would prefer to create a submodule `builtin_prefix` with `xml` and `xmlns` tests, and additional tests for overriding (maybe with TODOs that overriding rules should be changed)